Source/WebCore/dom/ClassCollection.h

```diff
@@ -65,6 +67,17 @@ inline bool ClassCollection::elementMatches(Element& element) const
         return false;
     if (m_classNames.isEmpty())
         return false;
+
+    auto& document = element.document();
+    if (element.document().inQuirksMode() || (document.decoder()->encoding().name() == "UTF-8"_s
```

Comment:
We most definitely don't want to be doing this check for every element. That's going to cause a perf regression.
We need to update the constructor where we set `SpaceSplitString::ShouldFoldCase::Yes` / `SpaceSplitString::ShouldFoldCase::No`.